client/extensions/woocommerce/app/promotions/fields/form-field.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+
+const FormField = ( {
+	labelText,
+	explanationText,
+	isRequired,
+	children,
+	translate,
+} ) => {
+	const requiredLabel = ( isRequired &&
+		<span className="fields__form-label-required">
+			{ translate( 'Required' ) }
+		</span>
+	);
+
+	const explanation = ( explanationText &&
+		<FormSettingExplanation>{ explanationText }</FormSettingExplanation>
+	);
+
+	return (
+		<FormFieldset>
+			<FormLabel>{ labelText }</FormLabel> { requiredLabel }
```

Comment:
A few accessibility notes:
- The label should be linked to its input with `htmlFor` (on the label) mapped to an `id` on the input
- The `requiredLabel` is part of the label, so it should be inside the `FormLabel` as well
- Would be nice to connect the `FormSettingExplanation` to the input, you can add an `id` to the `FormSettingExplanation`, and `aria-describedby` on the input to link the two.

I think if you just pass through `fieldName` you can create the ID with `${fieldName}-label`, `${fieldName}-description`, and that should be unique since you're already using it as the react key.

Reply:
Great feedback, thanks! I absent-mindedly thought this was something that `FormFieldset` already did.

Reply:
fieldsets don't actually do much for accessibility when used like this - they're meant to group related fields, with a `legend` a a group label. in this case it's basically a div.